datasets/librispeech_asr/librispeech_asr.py

```diff
@@ -0,0 +1,153 @@
+        "train.100": _DL_URL + "train-clean-100.tar.gz",
+        "train.360": _DL_URL + "train-clean-360.tar.gz",
+    },
+    "other": {
+        "test": _DL_URL + "test-other.tar.gz",
+        "dev": _DL_URL + "dev-other.tar.gz",
+        "train.500": _DL_URL + "train-other-500.tar.gz",
+    },
+}
+
+
+class LibrispeechASRConfig(datasets.BuilderConfig):
+    """BuilderConfig for LibriSpeechASR."""
+
+    def __init__(self, **kwargs):
+        """
+        Args:
+          data_dir: `string`, the path to the folder containing the files in the
+            downloaded .tar
+          citation: `string`, citation for the data set
+          url: `string`, url for information about the data set
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(LibrispeechASRConfig, self).__init__(version=datasets.Version("2.1.0", ""), **kwargs)
+
+
+class LibrispeechASR(datasets.GeneratorBasedBuilder):
+    """Librispeech dataset."""
+
+    BUILDER_CONFIGS = [
+        LibrispeechASRConfig(name="clean", description="'Clean' speech."),
+        LibrispeechASRConfig(name="other", description="'Other', more challenging, speech."),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "file": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                    "speaker_id": datasets.Value("int64"),
+                    "chapter_id": datasets.Value("int64"),
+                    "id": datasets.Value("string"),
+                }
+            ),
+            supervised_keys=("speech", "text"),
+            homepage=_URL,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        archive_path = dl_manager.download_and_extract(_DL_URLS[self.config.name])
+
+        if self.config.name == "clean":
+            train_splits = [
+                datasets.SplitGenerator(name="train.100", gen_kwargs={"archive_path": archive_path["train.100"]}),
+                datasets.SplitGenerator(name="train.360", gen_kwargs={"archive_path": archive_path["train.360"]}),
+            ]
+        elif self.config.name == "other":
+            train_splits = [
+                datasets.SplitGenerator(name="train.500", gen_kwargs={"archive_path": archive_path["train.500"]}),
+            ]
+
+        return train_splits + [
+            datasets.SplitGenerator(name=datasets.Split.VALIDATION, gen_kwargs={"archive_path": archive_path["dev"]}),
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs={"archive_path": archive_path["test"]}),
+        ]
+
+    def _generate_examples(self, archive_path):
+        """Generate examples from a Librispeech archive_path."""
+        transcripts_glob = os.path.join(archive_path, "LibriSpeech", "*/*/*/*.txt")
+        for transcript_file in glob.glob(transcripts_glob):
```

Comment:
```suggestion
        for transcript_file in sorted(glob.glob(transcripts_glob)):
```
we need to make the order deterministic here